packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx

```diff
@@ -599,6 +610,10 @@ export class ContextualMenu extends BaseComponent<IContextualMenuProps, IContext
         hasMenu: true
       };
     }
+    const btnRef = createRef<HTMLButtonElement>();
```

Comment:
We really should avoid using refs whenever possible. Especially not in the way that this one is used. Instead of passing a ref down, pass a queryselector through or something like that. If anything the item itself should decide what the target is.

Reply:
The ref is required so that the submenu positions itself correctly on the containing <button> and <a> elements (which is what they normally position on) which aren't available in ContextualMenuItem. An alternative approach is to move out the button and anchor items into their own component, like ContextualMenuSplitButton was moved. This would remove the need to pass down a ref (in the code you can see that I don't pass down that ref to ContextualMenuSplitButton). Again curious to hear @dzearing and @cliffkoh opinions

Reply:
Also up for discussion: passing the componentRef to ContextualMenuItem through IContextualMenuItem and 'renderItemRef' in order to set the componentRef on ContextualMenuItem. Again pinging @dzearing and @cliffkoh

Reply:
@kelseyyoung the best sounding approach would be to split button and anchor into their own components ( ContextualMenuButton and ContextualMenuAnchor) to remove the need to pass down a ref